drivers/pca9633/pca9633.c

```diff
@@ -0,0 +1,257 @@
+    assert(params);
+
+    dev->params = *params;
+
+    i2c_init(dev->params.i2c_dev);
+
+    int rc = _write_reg(dev, PCA9633_REG_MODE1, 0x0);
+    _write_reg(dev, PCA9633_REG_MODE2, 0x0);
+
+    if (rc != PCA9633_OK) {
+        return rc;
+    }
+
+    pca9633_set_ldr_state_all(dev, PCA9633_LDR_STATE_IND_GRP);
+    pca9633_set_group_control_mode(dev, PCA9633_GROUP_CONTROL_MODE_DIMMING);
+    pca9633_set_rgb(dev, 255, 255, 255);
+
+    return PCA9633_OK;
+}
+
+void pca9633_wakeup(pca9633_t* dev)
+{
+    uint8_t reg;
+    _read_reg(dev, PCA9633_REG_MODE1, &reg);
+    reg = reg & ~(1 << PCA9633_BIT_SLEEP);
+
+    _write_reg(dev, PCA9633_REG_MODE1, reg);
+}
+
+void pca9633_sleep(pca9633_t* dev)
+{
+    uint8_t reg;
+    _read_reg(dev, PCA9633_REG_MODE1, &reg);
+    reg = reg | (1 << PCA9633_BIT_SLEEP);
+
+    _write_reg(dev, PCA9633_REG_MODE1, reg);
+}
+
+void pca9633_turn_on(pca9633_t* dev)
+{
+    _write_reg(dev, PCA9633_REG_LEDOUT, dev->stored_reg_ledout);
+}
+
+void pca9633_turn_off(pca9633_t* dev)
+{
+    _read_reg(dev, PCA9633_REG_LEDOUT, &dev->stored_reg_ledout);
+    _write_reg(dev, PCA9633_REG_LEDOUT, PCA9633_LDR_STATE_OFF);
+}
+
+void pca9633_set_pwm(pca9633_t* dev, uint8_t reg_pwm, uint8_t pwm)
+{
+    _write_reg(dev, reg_pwm, pwm);
+}
+
+void pca9633_set_grp_pwm(pca9633_t* dev, uint8_t pwm)
+{
+    _write_reg(dev, PCA9633_REG_GRPPWM, pwm);
+}
+
+void pca9633_set_blinking(pca9633_t* dev, uint8_t blink_period,
+        uint8_t on_off_ratio)
+{
+    _write_reg(dev, PCA9633_REG_GRPFREQ, blink_period);
+    _write_reg(dev, PCA9633_REG_GRPPWM, on_off_ratio);
+}
+
+void pca9633_set_rgb(pca9633_t* dev, uint8_t r, uint8_t g, uint8_t b)
+{
+    pca9633_set_pwm(dev, dev->params.reg_pwm_red, r);
+    pca9633_set_pwm(dev, dev->params.reg_pwm_green, g);
+    pca9633_set_pwm(dev, dev->params.reg_pwm_blue, b);
+}
+
+void pca9633_set_rgba(pca9633_t* dev, uint8_t r, uint8_t g, uint8_t b, uint8_t a)
+{
+    pca9633_set_rgb(dev, r, g, b);
+
+    if (dev->params.has_amber_channel) {
+        pca9633_set_pwm(dev, dev->params.reg_pwm_amber, a);
+    }
+}
+
+void pca9633_set_ldr_state(pca9633_t* dev, uint8_t state, uint8_t ldr_bit)
+{
+    uint8_t reg;
+    _read_reg(dev, PCA9633_REG_LEDOUT, &reg);
+
+    /* first clear both bits of ldr */
+    reg = reg & ~(0b11 << ldr_bit);
+
+    /* second set new state to specified ldr */
+    reg |= (state << ldr_bit);
+
+    _write_reg(dev, PCA9633_REG_LEDOUT, reg);
+}
+
+void pca9633_set_ldr_state_all(pca9633_t* dev, uint8_t state)
+{
+    uint8_t reg = (state << PCA9633_BIT_LDR3)
+                | (state << PCA9633_BIT_LDR2)
+                | (state << PCA9633_BIT_LDR1)
+                | (state << PCA9633_BIT_LDR0);
+
+    _write_reg(dev, PCA9633_REG_LEDOUT, reg);
+}
+
+void pca9633_set_auto_increment(pca9633_t* dev, uint8_t option)
+{
+    bool enabled, bit0, bit1;
+
+    switch (option) {
+
+        case PCA9633_AI_ALL:
+            enabled = true;
+            bit0 = false;
+            bit1 = false;
```

Comment:
Actually you could just do
```suggestion
            new_reg = (1 << PCA9633_BIT_AI2)
                    | (0 << PCA9633_BIT_AI1)
                    | (0 << PCA9633_BIT_AI0) );
```